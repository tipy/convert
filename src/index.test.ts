import { convertSnakeToCamel } from './index';

describe('convertSnakeToCamel function tests', () => {
  it('convert a snake_case object into camelCase object', () => {
    const object = {
      first_name: 'test',
      soft_skills: [{ name: 'flexibility' }, { name: 'communication' }],
    };
    expect(convertSnakeToCamel(object)).toMatchSnapshot();
  });

  it('convert a snake_case array into camelCase array', () => {
    const object = [{ name: 'flexibility' }, { name: 'communication' }];
    expect(convertSnakeToCamel(object)).toMatchSnapshot();
  });

  it('returns the same argument passed if it is not object or array', () => {
    const arg = 'test'
    expect(convertSnakeToCamel(arg)).toMatchSnapshot();
  });
});
