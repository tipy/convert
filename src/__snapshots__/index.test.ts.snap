// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertSnakeToCamel function tests convert a snake_case array into camelCase array 1`] = `
Array [
  Object {
    "name": "flexibility",
  },
  Object {
    "name": "communication",
  },
]
`;

exports[`convertSnakeToCamel function tests convert a snake_case object into camelCase object 1`] = `
Object {
  "firstName": "test",
  "softSkills": Array [
    Object {
      "name": "flexibility",
    },
    Object {
      "name": "communication",
    },
  ],
}
`;

exports[`convertSnakeToCamel function tests returns the same argument passed if it is not object or array 1`] = `"test"`;
